src/checker/checker.rs

```diff
@@ -0,0 +1,372 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+/// Checker runs a set of consistency checks as materialized views. It spawns a separate
+/// thread per check. It will record how many times a check failed vs succeeded, along with
+/// the incorrect value.
+use std::ascii;
+use std::cmp::min;
+use std::collections::HashMap;
+use std::process;
+use std::str;
+use std::time::{Duration, Instant};
+
+use chrono::Utc;
+use env_logger::{Builder as LogBuilder, Env, Target};
+use log::{debug, error, info, warn};
+use pgrepr::{Interval, Numeric};
+use postgres::{Client, Row};
+use postgres_types::Type;
+use std::thread::{self, JoinHandle};
+
+use crate::args::{Args, Check, Source};
+
+mod args;
+
+type Error = Box<dyn std::error::Error + Send + Sync>;
+
+static MAX_BACKOFF: Duration = Duration::from_secs(2);
+
+fn main() -> Result<(), Error> {
+    LogBuilder::from_env(Env::new().filter_or("MZ_LOG", "info"))
+        .target(Target::Stdout)
+        .init();
+
+    let config = Args::from_cli()?;
+    info!("startup {}", Utc::now());
+
+    let init_result = initialize(&config);
+    // Start peek timing loop. This should never return.
+    if config.only_initialize {
+        match init_result {
+            Ok(()) => Ok(()),
+            Err(e) => {
+                error!("{}", e);
+                process::exit(1);
+            }
+        }
+    } else {
+        let _ = init_result;
+        // runtime.enter(|| run_checks(&config));
+        run_checks(&config);
+        unreachable!()
```

Comment:
This runs forever! Which totally works, but will probably bring up the "when am I done?" question for people. 

Would it make sense to add some sort of time limit here? 1 hour, 12 hours, etc.? Also, that would allow us to add this to nightly tests/release tests, which I think could be really useful.

Reply:
Done